### ISSUE TITLE
Inconsistent interaction between weight and include_background arguments for DiceLoss and DiceCELoss, with misleading error message

### ISSUE THREAD

**Issue by Jesse-Phitidis** [Project-MONAI/MONAI#7729]
**Describe the bug**

For `DiceLoss` if `include_background=False` then the `weight` argument must be of length `num_classes - 1` (i.e. without the background). If this is not the case then a ValueError is thrown with the message:

"The length of the `weight` sequence should be the same as the number of classes. If `include_background=False`, the weight should not include the background category class 0."

For `DiceCELoss` the `weight` argument must include the background, regardless of the status of the `include_background` argument. This is because it is what is required for the pytorch CE loss implementation used internally. Also internally, `weight[1:]` is passed to `DiceLoss` if `include_background=False`. If we mistakenly try to use `DiceCELoss` like it is `DiceLoss` and pass `num_classes - 1` weights with `include_background=False` then `DiceLoss` gets one too few weights and we get the same value error as above. Now, this is confusing because without examining the traceback one might think "that is exactly what I have given to this `DiceCELoss` class".

As a bonus it would also be nice if any sequence of weights was allowed for `DiceCELoss` and internally it was converted to a tensor of floats as required for pytorch CE loss.

**To Reproduce**

```
import torch
from monai.losses import DiceCELoss, DiceLoss

inp = torch.rand((1, 4, 10, 10, 10))
target = torch.randint(0, 4, (1, 1, 10, 10, 10))

dice_loss = DiceLoss(include_background=False, to_onehot_y=True, weight=torch.tensor([1,1,1])) 
dice_ce_loss = DiceCELoss(include_background=False, to_onehot_y=True, weight=torch.tensor([1,1,1])) 

dice_loss(inp, target) # no error
dice_ce_loss(inp, target) # error
```

**Expected behavior**

`DiceCELoss` should also accept `num_classes - 1` weights when `include_background=False`, so as to be consistent in the MONAI framework. A zero could simply be prepended to the weights internally before passing to pytorch CE loss.

**Environment**

================================
Printing MONAI config...
================================
MONAI version: 1.3.0
Numpy version: 1.26.4
Pytorch version: 2.2.2
MONAI flags: HAS_EXT = False, USE_COMPILED = False, USE_META_DICT = False
MONAI rev id: 865972f7a791bf7b42efbcd87c8402bd865b329e
MONAI __file__: /home/<username>/anaconda3/envs/CVDSeg/lib/python3.11/site-packages/monai/__init__.py

Optional dependencies:
Pytorch Ignite version: NOT INSTALLED or UNKNOWN VERSION.
ITK version: NOT INSTALLED or UNKNOWN VERSION.
Nibabel version: 5.2.1
scikit-image version: NOT INSTALLED or UNKNOWN VERSION.
scipy version: 1.12.0
Pillow version: 10.2.0
Tensorboard version: NOT INSTALLED or UNKNOWN VERSION.
gdown version: NOT INSTALLED or UNKNOWN VERSION.
TorchVision version: 0.17.2
tqdm version: 4.66.2
lmdb version: NOT INSTALLED or UNKNOWN VERSION.
psutil version: 5.9.0
pandas version: 2.1.1
einops version: NOT INSTALLED or UNKNOWN VERSION.
transformers version: NOT INSTALLED or UNKNOWN VERSION.
mlflow version: NOT INSTALLED or UNKNOWN VERSION.
pynrrd version: NOT INSTALLED or UNKNOWN VERSION.
clearml version: NOT INSTALLED or UNKNOWN VERSION.

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies

================================
Printing system config...
================================
System: Linux
Linux version: Ubuntu 22.04.4 LTS
Platform: Linux-6.5.0-28-generic-x86_64-with-glibc2.35
Processor: x86_64
Machine: x86_64
Python version: 3.11.9
Process name: python
Command: ['python', '-c', 'import monai; monai.config.print_debug_info()']
Open files: [popenfile(path='/home/jesse/.vscode-server/data/logs/20240430T064354/remoteagent.log', fd=19, position=491, mode='a', flags=33793), popenfile(path='/home/jesse/.vscode-server/data/logs/20240430T064354/ptyhost.log', fd=20, position=2014, mode='a', flags=33793), popenfile(path='/home/jesse/.vscode-server/data/logs/20240430T064354/network.log', fd=26, position=0, mode='a', flags=33793)]
Num physical CPUs: 32
Num logical CPUs: 64
Num usable CPUs: 64
CPU usage (%): [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.8, 46.1, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.7, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 2.6, 0.0, 0.0, 0.0, 25.9, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 3.4, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.9, 0.0, 0.0, 0.0, 22.6]
CPU freq. (MHz): 2
Load avg. in last 1, 5, 15 mins (%): [0.0, 0.1, 0.1]
Disk usage (%): 59.5
Avg. sensor temp. (Celsius): UNKNOWN for given OS
Total physical memory (GB): 251.5
Available memory (GB): 244.4
Used memory (GB): 5.0

================================
Printing GPU config...
================================
Num GPUs: 1
Has CUDA: True
CUDA version: 12.1
cuDNN enabled: True
NVIDIA_TF32_OVERRIDE: None
TORCH_ALLOW_TF32_CUBLAS_OVERRIDE: None
cuDNN version: 8902
Current device: 0
Library compiled for CUDA architectures: ['sm_50', 'sm_60', 'sm_61', 'sm_70', 'sm_75', 'sm_80', 'sm_86', 'sm_90']
GPU 0 Name: NVIDIA RTX A6000
GPU 0 Is integrated: False
GPU 0 Is multi GPU board: False
GPU 0 Multi processor count: 84
GPU 0 Total memory (GB): 47.5
GPU 0 CUDA capability (maj.min): 8.6
